sr_utilities_common/test/conditional_delayed_rostool_test.py

```diff
@@ -0,0 +1,133 @@
+from sr_utilities_common.conditional_delayed_rostool import wait_for_conditions
+
+class ConditionalDelayedRosToolTestCase(unittest.TestCase):
+    @classmethod
+    def setUp(self):
+        rospy.wait_for_message("test_mocap_topic", String)
+
+    @classmethod
+    def tearDown(self):
+        pass
+
+    def test_topic_element_type_is_valid(self):
+        element_type = "topic"
+        topics_list = ["test_mocap_topic"]
+        ros_topic_handler_class = RosElementsHandler(element_type, topics_list)
+        result = ros_topic_handler_class._retrieve_available_elements("topic")
+        self.assertIsNotNone(result)
+
+    def test_param_element_type_is_valid(self):
+        element_type = "param"
+        params_list = ["test_mocap_param"]
+        ros_param_handler_class = RosElementsHandler(element_type, params_list)
+        result = ros_param_handler_class._retrieve_available_elements(element_type)
+        self.assertIsNotNone(result)
+
+    def test_service_element_type_is_valid(self):
+        element_type = "service"
+        services_list = ["test_mocap_topic"]
+        ros_service_handler_class = RosElementsHandler(element_type, services_list)
+        result = ros_service_handler_class._retrieve_available_elements(element_type)
+        self.assertIsNotNone(result)
+
+    def test_invalid_element_type(self):
+        element_type = "invalid_element_type"
+        services_list = ["test_mocap_topic"]
+        self.ros_topic_handler_class = RosElementsHandler(element_type, services_list)
+        result = self.ros_topic_handler_class._retrieve_available_elements(element_type)
+        self.assertIsNone(result)
+
+    def test_check_if_required_element_is_published_list(self):
+        required_topics_list = ["/test_mocap_topic"]
+        ros_topic_handler_class = RosElementsHandler("topic", required_topics_list)
+        all_elements_available = ros_topic_handler_class.check_if_required_element_is_published()
+        self.assertEquals(all_elements_available, True)
+
+    def test_check_if_required_element_is_not_published_list(self):
+        required_topics_list = ["non_published_test_topic"]
+        ros_topic_handler_class = RosElementsHandler("topic", required_topics_list)
+        all_elements_available = ros_topic_handler_class.check_if_required_element_is_published()
+        self.assertEquals(all_elements_available, False)
+
+    def test_check_if_element_is_available_empty_list(self):
+        required_topics_list = []
+        ros_topic_handler_class = RosElementsHandler("topic", required_topics_list)
+        all_elements_available = ros_topic_handler_class.check_if_required_element_is_published()
+        self.assertEquals(all_elements_available, True)
+
+    def test_check_if_elements_is_available_empty_string(self):
+        required_topics_list=[""]
+        ros_topic_handler_class = RosElementsHandler("topic", required_topics_list)
+        try: 
```

Comment:
use assertRises
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises

Reply:
feels quite ugly to add to add a separate lambda funciton to assert the Raise https://ongspxm.gitlab.io/blog/2016/11/assertraises-testing-for-errors-in-unittest/

I'd prefer to keep it as it is, what do you think?

Reply:
no.